src/GSF.CarbonAware/src/Handlers/ForecastHandler.cs

```diff
@@ -88,16 +88,16 @@ private static EmissionsForecast ProcessAndValidateForecast(global::CarbonAware.
         var windowSize = parameters.Duration;
         var firstDataPoint = forecast.ForecastData.First();
         var lastDataPoint = forecast.ForecastData.Last();
-        forecast.DataStartAt = parameters.GetStartOrDefault(firstDataPoint.Time);
-        forecast.DataEndAt = parameters.GetEndOrDefault(lastDataPoint.Time + lastDataPoint.Duration);
-        forecast.Validate();
-        forecast.ForecastData = IntervalHelper.FilterByDuration(forecast.ForecastData, forecast.DataStartAt, forecast.DataEndAt);
-        forecast.ForecastData = forecast.ForecastData.RollingAverage(windowSize, forecast.DataStartAt, forecast.DataEndAt);
+        var dataStartAt = parameters.GetStartOrDefault(firstDataPoint.Time);
+        var dataEndAt = parameters.GetEndOrDefault(lastDataPoint.Time + lastDataPoint.Duration);
+        forecast.Validate(dataStartAt, dataEndAt);
+        forecast.ForecastData = IntervalHelper.FilterByDuration(forecast.ForecastData, dataStartAt, dataEndAt);
+        forecast.ForecastData = forecast.ForecastData.RollingAverage(windowSize, dataStartAt, dataEndAt);
         forecast.OptimalDataPoints = CarbonAwareOptimalEmission.GetOptimalEmissions(forecast.ForecastData);
-        if (forecast.ForecastData.Any())
+        /*if (forecast.ForecastData.Any())
```

Comment:
Is this not relevant anymore? Do we know if the` Duration` is set in case it is coming from the backend? Or is this another case of 'metadata' user driven?

Reply:
Deleted the commented code. Yes, it is not required since the window size comes from the ForecastData which is EmissionsData object and its already populated there.

Reply:
Thanks for the catch. I verified again and windowSize was not populated correctly. Fixed it.